src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NpgService.kt

```diff
@@ -0,0 +1,96 @@
+package it.pagopa.ecommerce.eventdispatcher.services.v2
+
+import it.pagopa.ecommerce.commons.documents.v2.authorization.NpgTransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.documents.v2.authorization.TransactionGatewayAuthorizationData
+import it.pagopa.ecommerce.commons.domain.v2.pojos.BaseTransactionWithRequestedAuthorization
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationResultDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OperationTypeDto
+import it.pagopa.ecommerce.commons.generated.npg.v1.dto.OrderResponseDto
+import it.pagopa.ecommerce.eventdispatcher.exceptions.InvalidNPGResponseException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+
+@Service
+class NpgService(
+  private val authorizationStateRetrieverService: AuthorizationStateRetrieverService
+) {
+
+  private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+  fun getAuthorizationDataFromNpgOrder(
+    transaction: BaseTransactionWithRequestedAuthorization
+  ): Mono<TransactionGatewayAuthorizationData> {
+    return getNpgOrderStatus(transaction, authorizationStateRetrieverService).flatMap {
+      when (it) {
+        is NpgOrderRefunded -> {
```

Comment:
If from npg the order results as refunded then we do not proceed with the refunds

Reply:
In previous GET state implementation we cannot see if a transaction is refunded, with this implementation yes (and it's the main purpose of use this NPG api call). 
I would put here the transaction in REFUNDED state since it's refunded NPG side writing the REFUND_REQUESTED and REFUNDED events into event store and updating the transaction status.
What do you think?